server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -77,6 +78,11 @@ public final class SearchHit implements Writeable, ToXContentObject, Iterable<Do
     private static final float DEFAULT_SCORE = Float.NaN;
     private float score = DEFAULT_SCORE;
 
+    private static final int NO_RANK = -1;
+    private int rank = NO_RANK;
+
+    private RankResult rankResult;
```

Comment:
Rank should be part of `RankResult`

Reply:
I was wondering if we should only use `RankResult` for explain or not and just have a simple _rank otherwise? If not, I will remove rank and only have `RankResult`.

Reply:
I removed this completely for now. Based on the precedent already set, it seems like this information is better conveyed as part of an explanation in a follow up. Now only a single int for rank exists.